crates/wysiwyg/src/composer_model/hyperlinks.rs

```diff
@@ -87,6 +91,26 @@ where
         self.create_update_replace_all()
     }
 
+    fn add_http_scheme(&mut self, link: &mut S) {
+        //   let mut new_link = link.clone();
+        let string = link.to_string();
+        let str = string.as_str();
+
+        match Url::parse(str) {
+            Ok(_) => {}
+            Err(ParseError::RelativeUrlWithoutBase) => {
+                let is_email = EmailAddress::is_valid(str);
+
+                if is_email {
+                    link.insert(0, &S::from("mailto:"));
+                } else {
+                    link.insert(0, &S::from("https://"));
```

Comment:
Should we add `http://` (rather than `https://`) and let sites do their own redirection to `https`? From testing it, this is the way browser URL bars handle input without a scheme.

Reply:
https://github.com/matrix-org/matrix-rich-text-editor/pull/481#issuecomment-1383878338

Reply:
Sorry I hadn't seen this, but makes sense. Users can still manually specify `http://` if needed.